src/main/java/mekanism/client/render/item/BakedCustomItemModel.java

```diff
@@ -375,38 +377,44 @@ private void doRender(TransformType type) {
 
             return faceQuads;
         }
+        boolean doCustomRendering = true;
```

Comment:
Can you please provide some commentary here on why GLContext.getCapabilities will crash and how we're using that test to determine how to handle the render?

Also, a little concerned about doing this try/catch every render cycle...

Reply:
`GlStateManager.translate` calls `GL11.glTranslatef`
The code for this method starts as
`ContextCapabilities caps = GLContext.getCapabilities();`
If it fails to be able to get the capabilities, then it will throw a runtime exception there.

I was not able to figure out why GLContext sometimes has null capabilities, but from what I can tell it has something to do with CodeChickenLib having its own BlockRendererDispatcher. I BELIEVE this makes it so that when the custom rendering happens for mekanism (which renders some items as blocks), gets called by TConstruct to render it in the world, it flips out potentially because of already rendering the block itself and not knowing how to render it internally. Still not sure why it would have null capabilities though.

The reason I pulled this out is that then it is easier to keep track of if it was running before when we paused it. I can do a little testing if you would rather, and try to make the try/catch be put around the pause/resume renderer. That way it doesn't make an extra check to `getCapabilities()`, though it may be slightly more messy. Not sure if I would have to pop the matrix regardless when it crashes or not. The other option which I am much more open to is to only do that try/catch if allowEmissiveItems is set to true by forge. (As this is the only case where I have seen it crash)

Reply:
One other thing to note, which I forgot to mention while writing the description... That if we do add that allowEmissiveItems check, we have to bump the min req forge version to 2781